auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -704,4 +750,101 @@ private static void buildParameterizedTypeArguments(CodeBlock.Builder block, Typ
       block.add("$T.class", typeArg);
     }
   }
+
+  private static Optional<ExecutableElement> findDefaultBuilder(TypeElement autoValueClass, TypeElement builderClass) {
```

Comment:
Nit: I prefer using the type directly + `@Nullable` rather than Optional when it's just implementation details and not differentiating between `null` and `absent` (that is, the sort of "three" implications of Optionals). Same with the other find method below.